Convert all orientation operations to double precision where possible

imikejackson opened the pull request:
All LaueOps classes and subclasses have been converted to use double precision floating point where practically possible.

imikejackson committed:
Converting the Laue Ops classes to use double precision by default

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Updates for the templated MatrixMath class

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

mmarineBlueQuartz commented on Source/OrientationLib/LaueOps/OrthoRhombicOps.cpp:
```diff
@@ -956,10 +966,10 @@ UInt8ArrayType::Pointer OrthoRhombicOps::generateIPFTriangleLegend(int imageDim)
+      if(sumSquares > 1.0) // Outside unit circle
       {
-        color = 0xFFFFFFFF;
+        color = 0xFFFFFFF;
```
Malformed hex color.  You need that missing F for an even number forming ARGB / RGBA or to remove another one for RGB.

mmarineBlueQuartz commented on Source/Plugins/Reconstruction/ReconstructionFilters/MergeColonies.cpp:
```diff
@@ -399,58 +400,59 @@ int32_t MergeColonies::getSeed(int32_t newFid)
-      float angdiff3 = fabsf(w - 60.00f);
-      float axisdiff3 = acosf(fabsf(n1) * 1.0000f + fabsf(n2) * 0.0000f + fabsf(n3) * 0.0000f);
+      float angdiff3 = std::fabs(w - 60.00f);
+      float axisdiff3 = std::acos(std::fabs(n1) * 1.0000f + std::fabs(n2) * 0.0000f + std::fabs(n3) * 0.0000f);
```
Why are we spending processing power on std::fabs(n3) and std::fabs(n2) when we're multiplying by 0?

imikejackson replied:
ummmm... got nothing ....

imikejackson replied:
git log + git blame...

imikejackson committed:
Fix color constant and remove unnecessary multiplications in MergeColonies

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Rewrite the OrientationTransform class using more C++ like style

The original version was a direct port of the Fortran codes where the output variable
is passed in as an argument. This rewrite will implement a C++ style of interface
where something is returned instead. This will allow operations to be chained together.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Updating Hexagonal ops with some compiler error fixes

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Fix failing unit tests for Quaternion and Orientation Transformation

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Add missing include directive

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

imikejackson committed:
Remove test files at end of unit test where possible.

Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>
